Userland/Services/LoginServer/LoginWindow.cpp

```diff
@@ -36,6 +38,12 @@ LoginWindow::LoginWindow(GUI::Window* parent)
             on_submit();
     };
 
+    m_shutdown_button = *widget.find_descendant_of_type_named<GUI::Button>("shutdown");
+    m_shutdown_button->on_click = [](auto) {
+        auto power_state_switch_node = MUST(Core::System::open("/sys/firmware/power_state", O_WRONLY));
```

Comment:
I think it would be better to invoke `/bin/shutdown` rather than having power state logic in LoginServer. it will make it more portable, and less annoying to change the shutdown API.